framework/src/interfaces/Coupleable.C

```diff
@@ -2140,10 +2140,22 @@ Coupleable::coupledIndices(const std::string & var_name) const
   return coupledVectorHelper<unsigned int>(var_name, func);
 }
 
+VariableName
+Coupleable::coupledName(const std::string & var_name, unsigned int comp) const
+{
+  if (getVar(var_name, comp))
+    return getVar(var_name, comp)->name();
+  // For constants, we ll just return the constant as the name
+  else if (MooseUtils::parsesToReal(var_name))
+    return var_name;
+  else
+    mooseError("Variable '", var_name, "' does not exist, yet its coupled name is requested");
```

Comment:
You have access to the MooseObject pointer to the derived class that issues the failing call. Outputting its name could be helpful for the user.